### ISSUE TITLE
Lookup parties - MSISDN not found test case giving incorrect Headers

### ISSUE THREAD

**Issue by sri-miriyala** [mojaloop/project#883]
@sri-miriyala commented on [Sun Jul 28 2019](https://github.com/mojaloop/mojaloop/issues/235)

**Summary:**
When I call GET /parties /MSISDN on a MSISDN not added to the MSISDN oracle, I'm expecting a party ID not found error msg with "3200" error code with right set of Headers returned to payerfsp simulator. But payerfsp received incorrect Headers FSPIOP-Source, FSPIOP-Destination, FSPIOP-URI, FSPIOP-Method and incorrect error code and msg too.

The tasks below include the scenarios for ALS and other services - quoting-service and other central-services (central-ledger), where an error message is sent from the Switch or for other calls that originate in the Switch (FSPIOP-URI , FSPIOP-Method should be present, while FSPIOP-Signature shouldn't be).

**Tasks**
- [x] Remove routine that deletes `FSPIO-URI` and `FSPIOP-Method` headers
- [x] Ensure FSPIOP-URI header value (e.g PUT) is in uppercase
- [ ] Check the same issue for other central services: 
  - [x] quoting-service
  - ~central-ledger~ Not necessary
  - ~central-settlement~ Not crucial for now
  - [x] account-lookup-service
  -  ~ml-api-adapter~ - (@oderayi - Moved to https://github.com/mojaloop/project/issues/1032)
  -  ~bulk-api-adapter~ - (@oderayi - Moved to https://github.com/mojaloop/project/issues/1032)

**Expected Headers:**
1. FSPIOP-Source to be Switch since the msg is originated in Switch
2. FSPIOP-Destination to be payerfsp as payerfsp is receiving the msg
3. FSPIOP-URI to be "/parties /MSISDN /ID /error"
4. FSPIOP-Method to be "PUT"

**Actual Headers:**
1. FSPIOP-Source is payerfsp
2. FSPIOP-Destination did not return
3. FSPIOP-URI did not return
4. FSPIOP-Method did not return

Error code returned should be 3200 with msg Generic ID error provided by the client which is generic ID not found but received 3204 with msg Party with the provided identifier, identifier type, and optional sub id or type was not found.

**Logs from ALS:**
[d1648-account-lookup-service-6875d999b8-65pbj account-lookup-service] 2019-07-28T03:01:28.209Z - debug: participant endpoint url: http://dev1-simulator.mojaloop.live/payerfsp/parties/MSISDN/10000000/error for endpoint type FSPIOP_CALLBACK_URL_PARTIES_PUT_ERROR
[d1648-account-lookup-service-6875d999b8-65pbj account-lookup-service] 2019-07-28T03:01:28.209Z - debug: sendRequest::request {"url":"http://dev1-simulator.mojaloop.live/payerfsp/parties/MSISDN/10000000/error","method":"PUT","headers":{"connection":"close","x-request-id":"4835f952cd7f5aede1591be8e363e1c3","x-real-ip":"10.1.2.119","x-forwarded-for":"10.1.2.119","x-forwarded-host":"dev1-account-lookup.mojaloop.live","x-forwarded-port":"80","x-forwarded-proto":"http","x-original-uri":"/parties/MSISDN/10000000","x-scheme":"http","date":"Sun, 28 Jul 2019 03:01:28 GMT","fspiop-source":"payerfsp","content-type":"application/vnd.interoperability.parties+json;version=1.0","user-agent":"PostmanRuntime/7.15.2","accept-encoding":"gzip, deflate"},"data":{"errorInformation":{"errorCode":"3204","errorDescription":"Party with the provided identifier, identifier type, and optional sub id or type was not found.","extensionList":[{"key":"MSISDN","value":"10000000"}]}},"responseType":"json"}

---

@elnyry commented on [Mon Jul 29 2019](https://github.com/mojaloop/mojaloop/issues/235#issuecomment-515936155)

@sri-miriyala I agree the issue about the headers but the error code looks good to me.. I do not see it as a bug. If a scheme needs to abstract information, 3200 may be used, but I do not see it in ALS why it would be a concern (security or otherwise) .. In fact 3200 doesn't talk about what kind of ID it is (transfer ID ? quote ID? etc)

@rmothilal FYI

## Pull Requets
- [x] https://github.com/mojaloop/central-services-shared/pull/96
- [x] https://github.com/mojaloop/account-lookup-service/pull/61
- [x] https://github.com/mojaloop/central-services-shared/pull/119
- [x] https://github.com/mojaloop/account-lookup-service/pull/114
- [x] https://github.com/mojaloop/ml-api-adapter/pull/234

**Comment by oderayi** [comment]
Looks like this bug is not fixed. With the current release, running against DEV2, with MSISDN 00000000000 (should be missing), response (body) from the simulator is empty. Headers still missing.

**Comment by elnyry** [comment]
In this instance, the error is being sent fine, but the HTTP callback wasn't reaching the simulator.. Below are the logs from the simulator.. @sri-miriyala @amarmodus @oderayi @rmothilal 
```
2019-08-23T12:54:41.829Z - info: sendRequest::request {"url":"http://d2741-centralledger-service:80/participants/payerfsp","method":"GET","headers":{"Accept":"application/vnd.interoperability.participants+json;version=1.0","FSPIOP-Destination":"switch","Content-Type":"application/vnd.interoperability.participants+json;version=1.0","Date":"Fri, 23 Aug 2019 12:54:41 GMT","FSPIOP-Source":"switch"},"responseType":"json"}
2019-08-23T12:54:41.837Z - info: Success: sendRequest::response {"status":200,"statusText":"OK","headers":{},"config":{"headers":{}},"request":{},"data":{}}
2019-08-23T12:54:41.840Z - info: sendRequest::request {"url":"http://dev2-simulator.mojaloop.live/oracle/participants/MSISDN/27713803999","method":"GET","headers":{"connection":"close","x-request-id":"4863a0ad670a8b1b7674c1a56507a353","x-real-ip":"10.1.2.119","x-forwarded-for":"10.1.2.119","x-forwarded-host":"dev2-account-lookup.mojaloop.live","x-forwarded-port":"80","x-forwarded-proto":"http","x-original-uri":"/parties/MSISDN/27713803999","x-scheme":"http","content-type":"application/vnd.interoperability.parties+json;version=1.0","fspiop-source":"payerfsp","fspiop-destination":"payeefsp","date":"Thu, 24 Jan 2019 10:22:12 GMT","user-agent":"PostmanRuntime/7.15.2","cache-control":"no-cache","postman-token":"8e9c67cd-d41f-4ef3-a6ad-82f66797aaf8","accept-encoding":"gzip, deflate"},"responseType":"json"}
2019-08-23T12:54:41.862Z - info: Success: sendRequest::response {"status":200,"statusText":"OK","headers":{},"config":{"headers":{}},"request":{},"data":[]}
2019-08-23T12:54:41.862Z - info: participantEndpointCache::getEndpoint::endpointType - FSPIOP_CALLBACK_URL_PARTIES_PUT_ERROR
2019-08-23T12:54:41.862Z - info: [fsp=payerfsp] ~ participantEndpointCache::fetchEndpoints := Refreshing the cache for FSP: payerfsp
2019-08-23T12:54:41.863Z - info: sendRequest::request {"url":"http://d2741-centralledger-service:80/participants/payerfsp/endpoints","method":"GET","headers":{"Accept":"application/vnd.interoperability.participants+json;version=1.0","FSPIOP-Destination":"switch","Content-Type":"application/vnd.interoperability.participants+json;version=1.0","Date":"Fri, 23 Aug 2019 12:54:41 GMT","FSPIOP-Source":"switch"},"responseType":"json"}
2019-08-23T12:54:41.869Z - info: Success: sendRequest::response {"status":200,"statusText":"OK","headers":{},"config":{"headers":{}},"request":{},"data":[{},{},{},{},{},{},{},{},{},{},{},{},{},{}]}
2019-08-23T12:54:41.870Z - info: sendRequest::request {"url":"http://dev2-simulator.mojaloop.live/payerfsp/parties/MSISDN/27713803999/error","method":"PUT","headers":{"connection":"close","x-request-id":"4863a0ad670a8b1b7674c1a56507a353","x-real-ip":"10.1.2.119","x-forwarded-for":"10.1.2.119","x-forwarded-host":"dev2-account-lookup.mojaloop.live","x-forwarded-port":"80","x-forwarded-proto":"http","x-original-uri":"/parties/MSISDN/27713803999","x-scheme":"http","content-type":"application/vnd.interoperability.parties+json;version=1.0","fspiop-source":"payerfsp","fspiop-destination":"payeefsp","date":"Thu, 24 Jan 2019 10:22:12 GMT","user-agent":"PostmanRuntime/7.15.2","cache-control":"no-cache","postman-token":"8e9c67cd-d41f-4ef3-a6ad-82f66797aaf8","accept-encoding":"gzip, deflate"},"data":{"errorInformation":{"errorCode":"3204","errorDescription":"Party not found","extensionList":{"extension":[{"key":"cause","value":"FSPIOPError\n at Object.createFSPIOPError (/opt/account-lookup-service/node_modules/@mojaloop/central-services-error-handling/"}]}}},"responseType":"json"}
2019-08-23T12:54:41.883Z - error: Request failed with status code 400
2019-08-23T12:54:41.884Z - error: Failed to send HTTP request to host
2019-08-23T12:54:41.884Z - error: Failed to send HTTP request to host
(node:1) UnhandledPromiseRejectionWarning: FSPIOPError: Failed to send HTTP request to host
at Object.createFSPIOPError (/opt/account-lookup-service/node_modules/@mojaloop/central-services-error-handling/src/factory.js:173:12)
at Object.sendRequest (/opt/account-lookup-service/node_modules/@mojaloop/central-services-shared/src/util/request.js:73:32)
at process._tickCallback (internal/process/next_tick.js:68:7)
Error: Request failed with status code 400
at createError (/opt/account-lookup-service/node_modules/axios/lib/core/createError.js:16:15)
at settle (/opt/account-lookup-service/node_modules/axios/lib/core/settle.js:17:12)
at IncomingMessage.handleStreamEnd (/opt/account-lookup-service/node_modules/axios/lib/adapters/http.js:237:11)
at IncomingMessage.emit (events.js:194:15)
at endReadableNT (_stream_readable.js:1125:12)
at process._tickCallback (internal/process/next_tick.js:63:19)
(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3825)
```

**Comment by oderayi** [comment]
@elnyry It appears these are logs for MSISDN `27713803999`. The scenario, as I understand and tested, is for MSISDN `00000000000`.

**Comment by oderayi** [comment]
Log for `00000000000`:

```JSON
2019-08-23T12:24:50.715Z - info: parties::getPartiesByTypeAndID::begin
2019-08-23T12:24:50.715Z - info: sendRequest::request {"url":"http://d2741-centralledger-service:80/participants/payerfsp","method":"GET","headers":{"Accept":"application/vnd.interoperability.participants+json;version=1.0","FSPIOP-Destination":"switch","Content-Type":"application/vnd.interoperability.participants+json;version=1.0","Date":"Fri, 23 Aug 2019 12:24:50 GMT","FSPIOP-Source":"switch"},"responseType":"json"}
2019-08-23T12:24:50.726Z - info: Success: sendRequest::response {"status":200,"statusText":"OK","headers":{},"config":{"headers":{}},"request":{},"data":{}}
2019-08-23T12:24:50.728Z - info: sendRequest::request {"url":"http://dev2-simulator.mojaloop.live/oracle/participants/MSISDN/00000000000","method":"GET","headers":{"connection":"close","x-request-id":"033fd04ab05184a02e99273a8218cbe4","x-real-ip":"10.1.2.119","x-forwarded-for":"10.1.2.119","x-forwarded-host":"dev2-account-lookup.mojaloop.live","x-forwarded-port":"80","x-forwarded-proto":"http","x-original-uri":"/parties/MSISDN/00000000000","x-scheme":"http","content-type":"application/vnd.interoperability.parties+json;version=1.0","fspiop-source":"payerfsp","fspiop-destination":"payeefsp","date":"Thu, 24 Jan 2019 10:22:12 GMT","user-agent":"PostmanRuntime/7.15.0","cache-control":"no-cache","postman-token":"91671cf1-cbe7-4adb-993a-19d372b0ac79","accept-encoding":"gzip, deflate"},"responseType":"json"}
2019-08-23T12:24:50.743Z - info: Success: sendRequest::response {"status":200,"statusText":"OK","headers":{},"config":{"headers":{}},"request":{},"data":[]}
2019-08-23T12:24:50.743Z - info: participantEndpointCache::getEndpoint::endpointType - FSPIOP_CALLBACK_URL_PARTIES_PUT_ERROR
2019-08-23T12:24:50.744Z - info: [fsp=payerfsp] ~ participantEndpointCache::fetchEndpoints := Refreshing the cache for FSP: payerfsp
2019-08-23T12:24:50.744Z - info: sendRequest::request {"url":"http://d2741-centralledger-service:80/participants/payerfsp/endpoints","method":"GET","headers":{"Accept":"application/vnd.interoperability.participants+json;version=1.0","FSPIOP-Destination":"switch","Content-Type":"application/vnd.interoperability.participants+json;version=1.0","Date":"Fri, 23 Aug 2019 12:24:50 GMT","FSPIOP-Source":"switch"},"responseType":"json"}
2019-08-23T12:24:50.752Z - info: Success: sendRequest::response {"status":200,"statusText":"OK","headers":{},"config":{"headers":{}},"request":{},"data":[{},{},{},{},{},{},{},{},{},{},{},{},{},{}]}
2019-08-23T12:24:50.753Z - info: sendRequest::request {"url":"http://dev2-simulator.mojaloop.live/payerfsp/parties/MSISDN/00000000000/error","method":"PUT","headers":{"connection":"close","x-request-id":"033fd04ab05184a02e99273a8218cbe4","x-real-ip":"10.1.2.119","x-forwarded-for":"10.1.2.119","x-forwarded-host":"dev2-account-lookup.mojaloop.live","x-forwarded-port":"80","x-forwarded-proto":"http","x-original-uri":"/parties/MSISDN/00000000000","x-scheme":"http","content-type":"application/vnd.interoperability.parties+json;version=1.0","fspiop-source":"payerfsp","fspiop-destination":"payeefsp","date":"Thu, 24 Jan 2019 10:22:12 GMT","user-agent":"PostmanRuntime/7.15.0","cache-control":"no-cache","postman-token":"91671cf1-cbe7-4adb-993a-19d372b0ac79","accept-encoding":"gzip, deflate"},"data":{"errorInformation":{"errorCode":"3204","errorDescription":"Party not found","extensionList":{"extension":[{"key":"cause","value":"FSPIOPError\n at Object.createFSPIOPError (/opt/account-lookup-service/node_modules/@mojaloop/central-services-error-handling/"}]}}},"responseType":"json"}
2019-08-23T12:24:50.763Z - error: Request failed with status code 400
2019-08-23T12:24:50.764Z - error: Failed to send HTTP request to host
2019-08-23T12:24:50.764Z - error: Failed to send HTTP request to host
(node:1) UnhandledPromiseRejectionWarning: FSPIOPError: Failed to send HTTP request to host
at Object.createFSPIOPError (/opt/account-lookup-service/node_modules/@mojaloop/central-services-error-handling/src/factory.js:173:12)
at Object.sendRequest (/opt/account-lookup-service/node_modules/@mojaloop/central-services-shared/src/util/request.js:73:32)
at process._tickCallback (internal/process/next_tick.js:68:7)
Error: Request failed with status code 400
at createError (/opt/account-lookup-service/node_modules/axios/lib/core/createError.js:16:15)
at settle (/opt/account-lookup-service/node_modules/axios/lib/core/settle.js:17:12)
at IncomingMessage.handleStreamEnd (/opt/account-lookup-service/node_modules/axios/lib/adapters/http.js:237:11)
at IncomingMessage.emit (events.js:194:15)
at endReadableNT (_stream_readable.js:1125:12)
at process._tickCallback (internal/process/next_tick.js:63:19)
(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3458)
```

**Comment by elnyry** [comment]
@oderayi 27713803999 is the ID that I tried this with

**Comment by oderayi** [comment]
With the current version, this is the content of the request looks like this:
```JSON
{
    "url": "http://172.17.0.1:8444/payerfsp/parties/MSISDN/27713803999/error",
    "method": "PUT",
    "headers": {
        "content-type": "application/vnd.interoperability.parties+json;version=1.0",
        "fspiop-source": "payerfsp",
        "fspiop-destination": "payeefsp",
        "date": "Fri, 06 Sep 2019 08:10:03 GMT",
        "user-agent": "PostmanRuntime/7.15.0",
        "cache-control": "no-cache",
        "postman-token": "2dd69768-ba50-4c48-a994-542fa3ffdd60",
        "accept-encoding": "gzip, deflate",
        "connection": "keep-alive"
    },
    "data": {
        "errorInformation": {
            "errorCode": "3204",
            "errorDescription": "Party not found",
            "extensionList": {
                "extension": [
                    {
                        "key": "cause",
                        "value": "FSPIOPError\n    at Object.createFSPIOPError (/home/steven/projects/modusbox/mojaloop/account-lookup-service/node_modules/@mojaloop/central-services-error-handling/src/factory.js:173:12)\n    at Object.getPartiesByTypeAndID (/home/steven/projects/modusbox/mojaloop/account-lookup-service/src/domain/parties/parties.js:64:32)\n    at process._tickCallback (internal/process/next_tick.js:68:7)"
                    }
                ]
            }
        }
    },
    "responseType": "json"
}
```

**Comment by elnyry** [comment]
Re-opened to add FSPIOP-URI and FSPIOP-Method headers to messages originating in the Switch (Eg: Error callbacks, additional notification to Payee, etc)

**Comment by oderayi** [comment]
Blocking story until https://github.com/mojaloop/project/issues/1030 is fixed.

**Comment by oderayi** [comment]
Unblocking issue because other remaining tasks have been moved to another issue.
